cmd/infra/powervs/create_test.go

```diff
@@ -0,0 +1,49 @@
+package powervs
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+)
+
+func TestUseExistingDHCP(t *testing.T) {
+	id1 := "id1"
+	id2 := "id2"
+
+	type expectedResult struct {
+		dhcpServerID string
+		err          error
+	}
+
+	tests := []struct {
+		input          models.DHCPServers
+		expectedResult expectedResult
+		name           string
+	}{
+		{
+			input:          models.DHCPServers{{ID: &id1}},
+			expectedResult: expectedResult{dhcpServerID: id1, err: nil},
+			name:           "DHCPServerDetail returned with no error",
+		},
+		{
+			input:          models.DHCPServers{{ID: &id1}, {ID: &id2}},
+			expectedResult: expectedResult{"", dhcpServerLimitExceeds(2)},
+			name:           "Error expected when more than one DHCPServer exist",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dhcpServerID, err := useExistingDHCP(test.input)
+			if !assert.Equal(t, dhcpServerID, test.expectedResult.dhcpServerID) {
+				t.Fatalf("unexpected dhcp server id received")
```

Comment:
no fatal is required, assert will take of that.

Reply:
removed the if.